core/templates/dev/head/pages/topic_editor/TopicEditorStateService.js

```diff
@@ -209,11 +212,43 @@ oppia.factory('TopicEditorStateService', [
        * _subtopicPages list.
        */
       setSubtopicPage: function(subtopicPage) {
-        if (_getSubtopicPageIndex(subtopicPage.getId()) !== -1) {
+        if (_getSubtopicPageIndex(subtopicPage.getId()) !== null) {
           _subtopicPages[_getSubtopicPageIndex(subtopicPage.getId())] =
             angular.copy(subtopicPage);
         } else {
           _setSubtopicPage(subtopicPage);
+          _newSubtopicPages.push(angular.copy(subtopicPage));
+        }
+      },
+
+      deleteSubtopicPage: function(topicId, subtopicId) {
+        var subtopicPageId = topicId + '-' + subtopicId.toString();
+        var index = _getSubtopicPageIndex(subtopicPageId);
+        var newIndex = _getNewSubtopicPageIndex(subtopicPageId);
+        if (index !== null) {
+          _subtopicPages.splice(index, 1);
+          // If the deleted subtopic page corresponded to a newly created
+          // subtopic, then the 'subtopicId' part of the id of all subsequent
+          // subtopic pages should be decremented to make it in sync with the
+          // their corresponding subtopic ids.
+          if (newIndex !== null) {
+            _newSubtopicPages.splice(newIndex, 1);
+            for (var i = 0; i < _subtopicPages.length; i++) {
+              var newSubtopicId = parseInt(_subtopicPages[i].getId().slice(13));
```

Comment:
Not clear where "13" comes from, and this sort of logic should be encapsulated in a common function throughout the frontend. It's too low-level to be written directly here, and there's already duplication several lines below.

Reply:
Ok, the 13 is actually the length of the topicId and the hyphen after it so as to extract just the subtopic id from the subtopic page id. Will write it in a function.

Reply:
Done.